### PR TITLE
Updated pull request template 

### PR DESCRIPTION
**Pull request by rt4914**
## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by rt4914** [comment]
@anandwana001 @BenHenning  @vinitamurthi  @seanlip Please have a look and leave your suggestions / reviews on this new updated PR template.

**Comment by BenHenning** [comment]
Thanks @rt4914. Left my first bit of feedback with @seanlip rather than here since he mentioned he'll be chatting with you later. Please assign back after next iteration.

**Comment by seanlip** [comment]
Reassigning to @rt4914 to update based on offline comments. @rt4914, please reassign me once you're ready for another pass -- thanks!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Let's maybe have these first few lines follow exactly lines 1-9 of https://github.com/oppia/oppia/edit/develop/.github/PULL_REQUEST_TEMPLATE.md

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Similarly, let's move the checklist up to here, and call it "Essential Checklist", similar to https://github.com/oppia/oppia/edit/develop/.github/PULL_REQUEST_TEMPLATE.md

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: Similarly, let's follow "Proof that changes are correct", but in the explanatory comment, you can add something about also explaining how the reviewer can verify the work. You can also say: for UI-related PRs, add screenshots and recordings (i.e. similar to what you wrote below) for both LTR and RTL layouts.

Also, I have a question. If the developer posts screenshots/recordings/etc., are the verification instructions still needed?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: This section should come last -- optional sections should be at the bottom. I would suggest phrasing it as:

```
## For UI-specific PRs only
<!-- Delete this section if this PR does not include UI-related changes. -->

### Espresso Test Screenshots
<!-- ... -->

### Accessibility checks
<!-- Explain how this PR has been tested for accessibility, or why doing so is not applicable. For more info, check out the [Oppia Android Accessibility Guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide). -->
```

(I moved RTL and screenshots/recordings to the "proof" section.)

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @rt4914! Left some notes.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Actually verifications instructions is important considering there are cases in app where we have to follow specific instructions to reach a certain screen. For example, opening a concept card, viewing hints and solution, etc.

Images and for proof that its correct and for very easy PR they help in merging the PR just by looking at the code and screenshots but for feature based PRs screenshots help in getting clear idea what to expect and instructions help in trying it out on own and cross verifying it in various situations like rotating the app, talkback, etc.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: 1 --> point 1

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: Just to check, do you use the "Reviewers" section? On the web repo, we use Assignees only because we don't have enough control over the other one.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: and in RTL mode --> in both LTR and RTL mode

**Review by seanlip — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [reply .github/PULL_REQUEST_TEMPLATE.md]: Ah ok. Is this only for UI-based PRs? If so, you can add a section at the bottom for that (it should probably come before the Espresso tests).

If not, then how is this section supposed to be filled in for non-UI-based PRs?

**Comment by seanlip** [comment]
(Also replied to https://github.com/oppia/oppia-android/pull/3664#discussion_r689340531 )

**Comment by BenHenning** [comment]
Another check that would be good to add:

```
- [ ] Any CI check exemptions have a rationale included in the PR description (i.e. any of these files have been changed: https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
```

(wording might need to some work)

We're trying to make sure changes to these files have reasons included.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Mostly it UI based PRs only. Like opening a concept card, etc.

Added at bottom 
```

### How to test this PR?
<!-- Explain what exact steps reviewer should perform to test this PR. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Considering now we have code-owners we actually don't need `Reviewers` section to be updated by contributor. 

Updated:
```
- [ ] The PR is **assigned** to appropriate reviewers. [Reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

**Comment by rt4914** [comment]
> Another check that would be good to add:
> 
> ```
> - [ ] Any CI check exemptions have a rationale included in the PR description (i.e. any of these files have been changed: https://github.com/oppia/oppia-android/tree/develop/scripts/assets)
> ```
> 
> (wording might need to some work)
> 
> We're trying to make sure changes to these files have reasons included.

Added this as 
```
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have rationale included in the PR description.
```

@BenHenning

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: Please provide a list of the exact steps that the reviewer should perform in order to test this PR.

**Review by seanlip — APPROVED** [review]
Just one note, otherwise LGTM. Thanks!

EDIT: scratch that, found a couple more. But otherwise it looks great!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: appropriate --> the appropriate

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: code changes --> auto-generated code changes (perhaps?)

Otherwise I was going to suggest linking to a wiki page explaining how to determine whether a change is unnecessary.

**Review by seanlip — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [inline .github/PULL_REQUEST_TEMPLATE.md]: @seanlip / @rt4914 I'm actually not keen on this. As someone who writes long & detailed descriptions, this makes me feel like I need to write shorter/more terse description which I don't actually think we want to encourage.

Why not keep the current Explanation section as-is? Point 1 is nice since it encourages people to mark PRs as fixing issues, but I rarely see this as an actual problem (people seem to pick up on it quickly), and it can be a nuisance for cases when we really don't have an issue to resolve (such as my recent codeowners PR). Plus, the checklist below already covers this requirement.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [reply .github/PULL_REQUEST_TEMPLATE.md]: I lean toward the wiki article--I doubt most people will know how to confidently check this box as-is.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [reply .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).
```

(also making Sean's change above since GitHub now only allows 1 comment per line of code).

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can we make this more actionable? E.g. maybe something like:

```
If your change includes PR changes, then:
- Add a video showing the full UX flow with a screen reader enabled (see accessibility guide)
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes (see screenshot guide)
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see RTL guide)
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing
```

This might need to be moved/adjusted in later sections, but the key difference is to be as actionable as possible & to provide context since some of these steps may be difficult or confusing for people not familiar with how to do them.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
<!-- Delete these sections if this PR does not include UI-related changes. -->
```

Got confused the first time I read this since I didn't realize it meant all of the UI-related sections.

**Review by BenHenning — COMMENTED** [review]
Thanks @rt4914! Left some initial thoughts building on what's been done so far.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [reply .github/PULL_REQUEST_TEMPLATE.md]: That's interesting -- the lack of "Fix #X" was a recurring problem on Web. I'm surprised to hear that it's not an issue on Android, but I defer to you on this and additionally agree that this part is not needed given the checkbox.

Given that, keeping the existing Explanation section seems reasonable to me. (Sorry @rt4914 for the back-and-forth on this.)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [reply .github/PULL_REQUEST_TEMPLATE.md]: Hm, @BenHenning, you mean "UI changes" and not "PR changes", right?

Note -- Espresso tests are covered in the section below under UI. I'm wondering whether we should move some of this stuff down there too (to the deletable section) if it's all UI-related. Do we want contributors of non-UI PRs to add any proof of correctness?

Otherwise, I agree with making things as actionable as possible.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: I have tried to make changes as per suggestion. Have a look and let me know your thoughts.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: Please fill in the explanation ...

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: rationale --> their rationale

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: PR changes --> UI-related changes

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: ...portrait/landscape for both the before & after UI changes in both a tablet and phone.

(Missing link to screenshot guide?)

**Review by seanlip — CHANGES_REQUESTED** [review]
Looks good to me, just grammar nits and some missing wiki links.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [reply .github/PULL_REQUEST_TEMPLATE.md]: Suggest moving this to be the first point since having a screenshot/video guide will be useful for all of the items I think.

**Review by BenHenning — COMMENTED** [review]
Thanks @rt4914! I think the latest version is looking really good. Just had 1 comment to add beyond Sean's.

Will take another pass once all the links/references are completed since they're really important to ensure the template is functioning as intended.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: starts --> each start

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (seanlip) [inline .github/PULL_REQUEST_TEMPLATE.md]: description --> explanation

**Review by seanlip — APPROVED** [review]
Thanks @rt4914! LGTM from my end, just two minor comments and the wiki page that Ben noted [here](https://github.com/oppia/oppia-android/pull/3664/files#r690991054) is left, I think.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (BenHenning) [reply .github/PULL_REQUEST_TEMPLATE.md]: Can we resolve this? In the very least, having a brief wiki page outlining what are construed as 'unnecessary changes' would be really helpful, I think.

**Review by BenHenning — COMMENTED** [review]
Thanks @rt4914. Just had one comment, otherwise it LGTM.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Created this new section here: https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rt4914) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

**Comment by rt4914** [comment]
@seanlip @BenHenning PTAL https://github.com/oppia/oppia-android/pull/3664#discussion_r694619874

**Review by seanlip — APPROVED** [review]
Thanks @rt4914! Looks great, no further comments.

@BenHenning could you PTAL?

**Review by BenHenning — APPROVED** [review]
Thanks @rt4914. LGTM!

**Comment by BenHenning** [comment]
Merging since everyone seems happy with this.

**Pull request merged**
